src/stacks-hierarchy/Root_ParkingViolationsReporting/components/SelectGroup.tsx

```diff
@@ -0,0 +1,139 @@
+import {Checkbox} from '@atb/components/checkbox';
+import {RadioIcon} from '@atb/components/radio';
+import {StyleSheet, useTheme} from '@atb/theme';
+import {useState} from 'react';
+import {
+  StyleProp,
+  TouchableOpacity,
+  View,
+  ViewProps,
+  ViewStyle,
+} from 'react-native';
+
+type SelectGroupProps<T> = ViewProps & {
+  items: T[];
+  multiple?: boolean;
+  onSelect: (selectedItems: T[]) => void;
+  renderItem: (item: T, isSelected: boolean) => React.ReactElement | null;
+  generateAccessibilityLabel?: (
+    item: T,
+    isSelected: boolean,
+  ) => string | undefined;
+  keyExtractor?: ((item: T, index: number) => string) | undefined;
+  itemStyle?: StyleProp<ViewStyle>;
+};
+
+export const SelectGroup = <T,>(props: SelectGroupProps<T>) => {
+  const [selectedItems, setSelectedItems] = useState<T[]>([]);
+  const {
+    items,
+    multiple = false,
+    onSelect,
+    renderItem,
+    generateAccessibilityLabel,
+    keyExtractor,
+    itemStyle,
+  } = props;
+
+  const onItemSelect = (item: T) => {
+    const newSelected = multiple
+      ? selectedItems.includes(item)
+        ? selectedItems.filter((it) => it !== item)
+        : [...selectedItems, item]
+      : [item];
+    setSelectedItems(newSelected);
+    onSelect(newSelected);
+  };
+
+  return (
+    <View {...props} accessibilityRole={multiple ? 'combobox' : 'radiogroup'}>
+      {items.map((item, i) => (
+        <SelectItem
+          style={itemStyle}
+          item={item}
+          selected={selectedItems.includes(item)}
+          multiple={multiple}
+          key={keyExtractor?.(item, i)}
+          onSelect={onItemSelect}
+          renderItem={renderItem}
+          generateAccessibilityLabel={generateAccessibilityLabel}
+        />
+      ))}
+    </View>
+  );
+};
+
+type ItemProps<T> = {
+  item: T;
+  multiple: boolean;
+  selected: boolean;
+  onSelect: (item: T) => void;
+  renderItem: (item: T, isSelected: boolean) => React.ReactElement | null;
+  generateAccessibilityLabel?: (
+    item: T,
+    isSelected: boolean,
+  ) => string | undefined;
+  style?: StyleProp<ViewStyle>;
+};
+const SelectItem = <T,>({
```

Comment:
```suggestion
const SelectItem = <T>({
```

Reply:
No. This is the way you have to do it to define generics without declaring a function using `function(){}`

Reply:
right. Looked like a comma that wasn't supposed to be there 😅